tests/python/pytorch/graphbolt/test_graphbolt_utils.py

```diff
@@ -184,6 +184,73 @@ def test_incomplete_unique_dst_nodes_():
         gb.unique_and_compact_node_pairs(node_pairs, unique_dst_nodes)
 
 
+def test_unique_and_compact_csc_formats_hetero():
+    N1 = torch.randint(0, 50, (30,))
+    N2 = torch.randint(0, 50, (20,))
+    N3 = torch.randint(0, 50, (10,))
+    unique_N1 = torch.unique(N1)
+    expected_unique_nodes = {
+        "n1": unique_N1,
+        "n2": N2,
+        "n3": N3,
+    }
+    csc_formats = {
+        "n1:e1:n2": gb.CSCFormatBase(
+            indptr=torch.range(0, 21),
+            indices=N1[:20],
+        ),
+        "n1:e2:n3": gb.CSCFormatBase(
+            indptr=torch.range(0, 11),
+            indices=N1[20:30],
+        ),
+        "n2:e3:n3": gb.CSCFormatBase(
+            indptr=torch.range(0, 11),
+            indices=N2[10:],
+        ),
+    }
+
+    dst_nodes = {
+        "n2": N2,
+        "n3": N3,
+    }
+
+    unique_nodes, compacted_csc_formats = gb.unique_and_compact_csc_formats(
+        csc_formats, dst_nodes
+    )
+    for ntype, nodes in unique_nodes.items():
+        expected_nodes = expected_unique_nodes[ntype]
+        assert torch.equal(torch.sort(nodes)[0], torch.sort(expected_nodes)[0])
+    for etype, pair in compacted_csc_formats.items():
+        indices = pair.indices
+        indptr = pair.indptr
+        indices_type, _, _ = gb.etype_str_to_tuple(etype)
+        indices = unique_nodes[indices_type][indices]
+        expected_indices = csc_formats[etype].indices
+        expected_indptr = csc_formats[etype].indptr
+        assert torch.equal(indices, expected_indices)
+        assert torch.equal(indptr, expected_indptr)
+
+
+def test_unique_and_compact_csc_formats_homo():
+    N = torch.cat((torch.arange(0, 10), torch.randint(10, 50, (190,))))
```

Comment:
I did not call it out in the test_unique_and_compact_homo, but using randint to create test cases is actually not a good practise. Better to just hard code the number such that it is easier to reproduce bug when test fails.